test/old_tests/UnitTests/array.cpp

```diff
@@ -1223,3 +1223,32 @@ TEST_CASE("array_view,compare,array_view")
     REQUIRE(result.greater_equal);
     REQUIRE(!result.less_equal);
 }
+
+// Verify that class template argument deduction works for array_view.
+TEST_CASE("array_view,ctad")
+{
+#define REQUIRE_DEDUCED_AS(T, ...) \
+    static_assert(std::is_same_v<array_view<T>, decltype(array_view(__VA_ARGS__))>)
+
+    uint8_t a[3]{};
+    REQUIRE_DEDUCED_AS(uint8_t, &a[0], &a[0]);
+    REQUIRE_DEDUCED_AS(uint8_t, a);
+
+    std::array<uint8_t, 3> ar{};
+    REQUIRE_DEDUCED_AS(uint8_t, ar);
+
```

Comment:
Please add a test case also for `std::array<uint8_t const, 3>`. `vector`-of-const isn't terribly useful in the real world, but `array`-of-const is.